modem/design.md

```diff
@@ -0,0 +1,8 @@
+Overview:
+
+![Design diagram](diagram.png)
+
+We create a Transceiver object that will manage the serial connection. It will do command detection and hand off to the appropriate Command object to handle the interaction. The Command object will be subclassed for each type of command we will receive and will handle the processing of the command and generating the response.
```

Comment:
How do you plan to dynamically dispatch the command to the right instance of `Command` object?

And what's the lifecycle of a `Command` object? Does it carry any state or is it purely a function wrapper?